adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileADALTests.cs

```diff
@@ -37,37 +40,131 @@ namespace Test.ADAL.NET.UIAutomation
+            controller.DismissKeyboard();
+
+            //Enter Resource
+            controller.EnterText(UiTestConstants.resourceEntryID, Resource, false);
+            controller.DismissKeyboard();
+        }
+
+        private static void PerformSignInFlow(ITestController controller, IUser user)
+        {
+            string passwordInputID = string.Empty;
+            string signInButtonID = string.Empty;
+
+            switch (user.FederationProvider)
+            {
+                case FederationProvider.AdfsV4:
+                    passwordInputID = UiTestConstants.AdfsV4WebPasswordID;
+                    signInButtonID = UiTestConstants.AdfsV4WebSubmitID;
+                    break;
+                default:
+                    passwordInputID = UiTestConstants.WebPasswordID;
+                    signInButtonID = UiTestConstants.WebSubmitID;
+                    break;
+            }
 
             //Acquire token flow
-            controller.Tap("acquireToken");
+            controller.Tap(UiTestConstants.AcquireTokenID);
             //i0116 = UPN text field on AAD sign in endpoint
-            controller.EnterText("i0116", user.Upn, true);
+            controller.EnterText(UiTestConstants.WebUPNInputID, 20, user.Upn, true);
             //idSIButton9 = Sign in button
-            controller.Tap("idSIButton9", true);
+            controller.Tap(UiTestConstants.WebSubmitID, true);
             //i0118 = password text field
-            controller.EnterText("i0118", ((LabUser)user).GetPassword(), true);
-            controller.Tap("idSIButton9", true);
+            controller.EnterText(passwordInputID, ((LabUser)user).GetPassword(), true);
+            controller.Tap(signInButtonID, true);
+        }
+
+        private static void VerifyResult(ITestController controller)
+        {
+            //There may be a delay in the amount of time it takes for an authentication request to complete.
+            //Thus this method will check the result once a second for 10 seconds.
+            bool done = false;
+            int attempts = 0;
+            int maximumAttempts = 20;
+
+            while (!done)
+            {
+                if (attempts > maximumAttempts)
+                    done = true;
+
+                attempts++;
+
+                //Test results are put into a label that is checked for messages
+                var result = controller.GetText(UiTestConstants.TestResultID);
+                if (result.Contains(UiTestConstants.TestResultSuccsesfulMessage))
+                {
+                    return;
+                }
+                else if (result.Contains(UiTestConstants.TestResultFailureMessage))
+                {
+                    Assert.Fail();
+                    return;
+                }
 
-            //Verify result. Test results are put into a label
-            Assert.IsTrue(controller.GetText("testResult") == "Success: True");
+				Thread.Sleep(1000);
```

Comment:
>Thread.Sleep(1000); [](start = 4, length = 19)

argh... I get really scared about Thread.Sleeps in the code....

Reply:
Actually it seems this is just a polling mechanism so a form of waiting is fine. Would be better if the app would use async / await and then you could wait without blocking.